bazel/foreign_cc/BUILD

```diff
@@ -231,66 +231,69 @@ envoy_cmake_external(
             "libLLVMDebugInfoPDB.a",
-            "libLLVMDebugInfoDWARF.a",
-            "libLLVMFuzzMutate.a",
-            "libLLVMTableGen.a",
-            "libLLVMDlltoolDriver.a",
-            "libLLVMLineEditor.a",
-            "libLLVMOrcJIT.a",
-            "libLLVMCoverage.a",
+            "libLLVMLTO.a",
+            "libLLVMObjCARCOpts.a",
             "libLLVMMIRParser.a",
-            "libLLVMObjectYAML.a",
-            "libLLVMLibDriver.a",
-            "libLLVMOption.a",
-            "libLLVMWindowsManifest.a",
+            "libLLVMOrcJIT.a",
+            "libLLVMOrcError.a",
+            "libLLVMJITLink.a",
+            "libLLVMPasses.a",
+            "libLLVMipo.a",
+            "libLLVMInstrumentation.a",
+            "libLLVMVectorize.a",
+            "libLLVMLinker.a",
+            "libLLVMIRReader.a",
+            "libLLVMAsmParser.a",
             "libLLVMX86Disassembler.a",
             "libLLVMX86AsmParser.a",
             "libLLVMX86CodeGen.a",
+            "libLLVMCFGuard.a",
             "libLLVMGlobalISel.a",
             "libLLVMSelectionDAG.a",
             "libLLVMAsmPrinter.a",
-            "libLLVMDebugInfoCodeView.a",
-            "libLLVMDebugInfoMSF.a",
+            "libLLVMDebugInfoDWARF.a",
+            "libLLVMCodeGen.a",
+            "libLLVMScalarOpts.a",
+            "libLLVMInstCombine.a",
+            "libLLVMAggressiveInstCombine.a",
+            "libLLVMTransformUtils.a",
+            "libLLVMBitWriter.a",
             "libLLVMX86Desc.a",
             "libLLVMMCDisassembler.a",
-            "libLLVMX86Info.a",
             "libLLVMX86Utils.a",
+            "libLLVMX86Info.a",
             "libLLVMMCJIT.a",
-            "libLLVMInterpreter.a",
             "libLLVMExecutionEngine.a",
-            "libLLVMRuntimeDyld.a",
-            "libLLVMCodeGen.a",
             "libLLVMTarget.a",
-            "libLLVMCoroutines.a",
-            "libLLVMipo.a",
-            "libLLVMInstrumentation.a",
-            "libLLVMVectorize.a",
-            "libLLVMScalarOpts.a",
-            "libLLVMLinker.a",
-            "libLLVMIRReader.a",
-            "libLLVMAsmParser.a",
-            "libLLVMInstCombine.a",
-            "libLLVMTransformUtils.a",
-            "libLLVMBitWriter.a",
             "libLLVMAnalysis.a",
             "libLLVMProfileData.a",
+            "libLLVMRuntimeDyld.a",
             "libLLVMObject.a",
+            "libLLVMTextAPI.a",
             "libLLVMMCParser.a",
-            "libLLVMMC.a",
             "libLLVMBitReader.a",
-            "libLLVMBitstreamReader.a",
+            "libLLVMMC.a",
+            "libLLVMDebugInfoCodeView.a",
+            "libLLVMDebugInfoMSF.a",
             "libLLVMCore.a",
+            "libLLVMRemarks.a",
+            "libLLVMBitstreamReader.a",
             "libLLVMBinaryFormat.a",
             "libLLVMSupport.a",
             "libLLVMDemangle.a",
-            "libLLVMRemarks.a",
-            "libLLVMCFGuard.a",
-            "libLLVMTextAPI.a",
```

Comment:
Also, what's the source of this new order? It doesn't match the output from:
```
docker run -it --rm envoyproxy/envoy-build-ubuntu:3d0491e2034287959a292806e3891fd0b7dd2703 /bin/bash -c '/opt/llvm/bin/llvm-config --libnames'
```
I think this wasn't updated since LLVM 6.0, and somehow it didn't break, but I believe that the order should match the output from the command above.

Reply:
My teammate works for this part and he is on vacation now, may responses slowly for it.

Reply:
Yes. The reordering is necessary for WAMR. The previous one leads an undefined symbol linking error when enabling WAMR. 
We also run a test to enable WAVM and it works well with the reordering list

Reply:
That's great, thanks for testing. It sounds like the reordering is justified.

However, 2 questions still remain unanswered:
1) What's the source of this new order?
2) Does the order produced by `llvm-config --libnames` from LLVM 10 also work?

Thanks!